### PR TITLE
Multienv support in the transform.

### PR DESCRIPTION
**Pull request by mikeparisstuff**
*Description of changes:*

This adds support for multiple environments in the transform.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/graphql-dynamodb-transformer/src/resources.ts [hunk 1]
```diff
@@ -145,7 +157,15 @@ export class ResourceFactory {
      */
     public makeIAMRole(tableId: string) {
         return new IAM.Role({
-            RoleName: Fn.Join('-', [tableId, 'role', Fn.GetAtt(ResourceConstants.RESOURCES.GraphQLAPILogicalID, 'ApiId')]),
+            RoleName: Fn.If(
+                ResourceConstants.CONDITIONS.HasEnvironmentParameter,
+                Fn.Join('-', [
```
Comment (kaustavghosh06) [inline packages/graphql-dynamodb-transformer/src/resources.ts]: Doesn't roles have a limitation of 64 chars as well? This name seems to be fragile.

File: packages/graphql-dynamodb-transformer/src/resources.ts [hunk 2]
```diff
@@ -160,7 +180,20 @@ export class ResourceFactory {
             },
             Policies: [
                 new IAM.Role.Policy({
-                    PolicyName: 'DynamoDBAccess',
+                    PolicyName: Fn.If(
```
Comment (kaustavghosh06) [inline packages/graphql-dynamodb-transformer/src/resources.ts]: You don't need to modify policy names. Only role names can't clash. There can be policies with the same name under different IAM roles.

File: packages/graphql-elasticsearch-transformer/src/resources.ts [hunk 3]
```diff
@@ -348,54 +334,37 @@ export class ResourceFactory {
...
-                "-",
+    private domainName() {
+        return Fn.If(
+            ResourceConstants.CONDITIONS.HasEnvironmentParameter,
+            Refs.NoValue,
+            Fn.Join(
+                '-',
                 [
-                    "d",
+                    'd',
```
Comment (kaustavghosh06) [inline packages/graphql-elasticsearch-transformer/src/resources.ts]: What is this for?

File: packages/graphql-dynamodb-transformer/src/resources.ts (hunk 2 above)
Comment (mikeparisstuff) [reply packages/graphql-dynamodb-transformer/src/resources.ts]: Makes sense. I'll revert this.

File: packages/graphql-elasticsearch-transformer/src/resources.ts (hunk 3 above)
Comment (mikeparisstuff) [reply packages/graphql-elasticsearch-transformer/src/resources.ts]: The api id is 26 characters so this was just to give the cluster a slightly different name than api id.

File: packages/graphql-dynamodb-transformer/src/resources.ts (hunk 1 above)
Comment (mikeparisstuff) [reply packages/graphql-dynamodb-transformer/src/resources.ts]: I have limited the length of the table id.

**Pull request merged**